src/main.c

```diff
@@ -397,146 +397,111 @@ static void apply_operations(numberstack* numbers, operation** current_op) {
 
-        if (inp == 127)
-        {
+        if (inp == 127) {
             // Backspace
 
-            if (pos != 0){
+            if (pos != 0) {
                 pos--;
                 len--;
+                inp = '\0';
+            }
+            else {
+                // Skip printing if backspace was pressed but nothing was done, otherwise a strange undefined character is printed
+                searched = 1;
             }
-            inp = '\0';
         }
 
-        if(!searched) {
-            // Prevent user to input more than MAX_IN
-            if (len <= MAX_IN && !browsing) {
+        // Prevent user to input more than MAX_IN
+        if(!searched && len <= MAX_IN) {
+            if (!browsing) {
                 // If the cursor is at the end of the text
                 
                 // Append char to in array
                 in[pos] = inp;
                 in[++pos] = '\0';
                 len++; // Make sure that len is still equal to pos
 
-                if (inp == '\0')
-                {
+                if (inp == '\0') {
                     // Clear screen from previous input
                     mvwprintw(inputwin, 1, 22 + --len ," ");
                 }
             }
-            else if (len < MAX_IN && browsing) {
+            else {
                 // If the cursor is in the text, not at the end
-                // len < MAX_IN is allowed because the underscore adds one to length and the underscore isn't really a part of in
 
-                if (inp == '\0'){
+                if (inp == '\0') {
                     // Backspace
                     
-                    // Check if pos is already at the start of the input, if so don't do anything
-                    if (in[0] != '_'){
-                        in[pos] = '_';
-
-                        // Move all of in after pos one space back
-                        for (int i = pos; i <= len; i++){
-                            in[i] = in[i + 1];
-                        }
-                        // Clear screen from previous input
-                        mvwprintw(inputwin, 1, 22 + len ," ");
+                    // Move all of in after pos one space back
+                    for (int i = pos; i <= len; i++) {
+                        in[i] = in[i + 1];
                     }
+                    // Clear screen from previous input
+                    mvwprintw(inputwin, 1, 22 + len ," ");
                 }
                 else {
                     // Everything except backspace
 
-                    // Append char to in array
-                    in[pos++] = inp;
-
-                    // Move all of in after pos one space forward to make room for the underscore
+                    // Move all of in after pos one space forward to make room for the new input
                     len++;
                     for (int i = len; i > pos; i--) {
                         in[i] = in[i - 1];
-                    }
+                     }
```

Comment:
The code looks fantastic, and I didn't test it but I'm sure you did and it's working.

I'm just going to point out 3 things for you to correct regarding the presentation (because I can't really patch it right now), and when you send this second commit I'll merge it

In this line I think there's 1 ` ` (space) too much